Source/Renderer/FramebufferManager.js

```diff
@@ -0,0 +1,354 @@
+        this._colorTextures[i] = new Texture({
+          context: context,
+          width: width,
+          height: height,
+          pixelFormat: pixelFormat,
+          pixelDatatype: pixelDatatype,
+          sampler: Sampler.NEAREST,
+        });
+      }
+    }
+
+    // Create depth stencil texture or renderbuffer
+    if (this._depthStencil && this._createDepthAttachments) {
+      if (this._supportsDepthTexture && context.depthTexture) {
+        this._depthStencilTexture = new Texture({
+          context: context,
+          width: width,
+          height: height,
+          pixelFormat: PixelFormat.DEPTH_STENCIL,
+          pixelDatatype: PixelDatatype.UNSIGNED_INT_24_8,
+          sampler: Sampler.NEAREST,
+        });
+      } else {
+        this._depthStencilRenderbuffer = new Renderbuffer({
+          context: context,
+          width: width,
+          height: height,
+          format: RenderbufferFormat.DEPTH_STENCIL,
+        });
+      }
+    }
+
+    // Create depth texture
+    if (this._depth && this._createDepthAttachments) {
+      if (this._supportsDepthTexture && context.depthTexture) {
+        this._depthTexture = new Texture({
+          context: context,
+          width: width,
+          height: height,
+          pixelFormat: PixelFormat.DEPTH_COMPONENT,
+          pixelDatatype: PixelDatatype.UNSIGNED_INT,
+          sampler: Sampler.NEAREST,
+        });
+      } else {
+        this._depthRenderbuffer = new Renderbuffer({
+          context: context,
+          width: width,
+          height: height,
+          format: RenderbufferFormat.DEPTH_COMPONENT16,
+        });
+      }
+    }
+
+    this._framebuffer = new Framebuffer({
+      context: context,
+      colorTextures: this._colorTextures,
+      depthTexture: this._depthTexture,
+      depthRenderbuffer: this._depthRenderbuffer,
+      depthStencilTexture: this._depthStencilTexture,
+      depthStencilRenderbuffer: this._depthStencilRenderbuffer,
+      destroyAttachments: false,
+    });
+  }
+};
+
+FramebufferManager.prototype.getColorTexture = function (index) {
+  index = defaultValue(index, 0);
+  //>>includeStart('debug', pragmas.debug);
+  if (index >= this._colorAttachmentsLength) {
+    throw new DeveloperError(
+      "index must be smaller than total number of color attachments."
+    );
+  }
+  //>>includeEnd('debug');
+  return this._colorTextures[index];
+};
+
+FramebufferManager.prototype.setColorTexture = function (texture, index) {
+  index = defaultValue(index, 0);
+  //>>includeStart('debug', pragmas.debug);
+  if (this._createColorAttachments) {
+    throw new DeveloperError(
+      "createColorAttachments must be false if setColorTexture is called."
+    );
+  }
+  if (index >= this._colorAttachmentsLength) {
+    throw new DeveloperError(
+      "index must be smaller than total number of color attachments."
+    );
+  }
+  //>>includeEnd('debug');
+  this._colorTextures[index] = texture;
+  this._attachmentsDirty = true;
```

Comment:
Maybe only set `attachmentsDirty` if the texture changed. That way if something sets the color texture to the same value every frame the framebuffer wouldn't be considered dirty. I don't think this is happening anywhere now but if it did happen in the future the code would stay fast.

```suggestion
  this._attachmentsDirty = texture !== this._colorTextures[index];
  this._colorTextures[index] = texture;
```

Reply:
Sounds good, I like that change. Any reason to only check this for color textures and not all the attachments?

Reply:
Ah I just commented on this one, but you could do the same thing for all the attachments

Reply:
Cool, I made that change in 26bf8ad